cmd/utils.go

```diff
@@ -187,23 +188,32 @@ func getLookupType(l string) minio.BucketLookupType {
 
 // struct representing object prefix and sse keys association.
 type prefixSSEPair struct {
-	prefix string
-	sseKey string
+	Prefix string
+	SSE    encrypt.ServerSide
 }
 
 // parse and validate encryption keys entered on command line
-func parseAndValidateEncryptionKeys(sseKeys string) (encMap map[string][]prefixSSEPair, err *probe.Error) {
-	if sseKeys == "" {
-		return
-	}
+func parseAndValidateEncryptionKeys(sseKeys string, sse string) (encMap map[string][]prefixSSEPair, err *probe.Error) {
 	encMap, err = parseEncryptionKeys(sseKeys)
 	if err != nil {
 		return nil, err
 	}
+	if sse != "" {
+		for _, prefix := range strings.Split(sse, ",") {
+			alias, _ := url2Alias(prefix)
+			if _, ok := encMap[alias]; !ok {
+				encMap[alias] = make([]prefixSSEPair, 0)
+			}
+			encMap[alias] = append(encMap[alias], prefixSSEPair{
+				Prefix: prefix,
+				SSE:    encrypt.NewSSE(),
+			})
+		}
+	}
```

Comment:
if mc cp/mirror specify both --encrypt and --encrypt-key and both prefixes match, it would produce inconsistent results 
In the below case, it gets SSE-C encrypted whereas an user might expect SSE-S3 encryption
``` 
mc cp /etc/issue myminio-ssl/tbucket/abcd/a --insecure  --encrypt-key="myminio-ssl/tbucket=$(echo original-password | openssl dgst -sha256 -binary | xxd -p -c 32 -l 16)" --encrypt "myminio-ssl/tbucket/abcd/" 
```

Since cp/mirror allows --recursive mode, and there might be a mixed bag of SSE-C/SSE-S3 encrypted objects we should reject cp/mirror of objects where the --encrypt and --encrypt-key share a matching alias/prefix to avoid confusion.